MapboxDirections/MBIntersection.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+public class Intersection: NSObject, NSSecureCoding {
+    public var inIndex: Int?
```

Comment:
Actually, to avoid confusion with the existing `exitIndex` property on RouteStep, how about `approachIndex` and `outletIndex`, or `ingressIndex` and `egressIndex`?